tsdb/engine/tsm1/cache.go

```diff
@@ -0,0 +1,124 @@
+package tsm1
+
+import (
+	"sync"
+)
+
+// entry is the set of all values received for a given key.
+type entry struct {
+	mu         sync.Mutex
+	values     Values // All stored values.
+	checkpoint uint64 // The checkpoint associated with the latest addition to values
+	unsorted   bool   // Whether the data requires sorting and deduping before query.
+}
+
+func newEntry() *entry {
+	return &entry{values: make(Values, 0)}
```

Comment:
You don't need to allocate here.  Appending to an initial slice state in `add` is fine.